### ISSUE TITLE
hledger-git: git: 'record' is not a git command

### ISSUE THREAD

**Issue by pfiaux** [simonmichael/hledger#1942]
I installed `hledger-git` and while looking through the script I noticed record calls `git record ...`.

https://github.com/simonmichael/hledger/blob/25c9083d0dcbed80498b227b46a3e3a12b17315d/bin/hledger-git#L72-L79

Should it call `git commit` instead on line 78?

To reproduce (with hledger-git installed):
```
⟩ hledger git record "test"
git: 'record' is not a git command. See 'git --help'.

The most similar command is
        restore
```
(same with `hledger-git record "test"`)

I'm on macOS, and I think hledger version isn't relevant here since the issue seems to be in the add on script.

**Comment by simonmichael** [comment]
Thanks for the report, you're quite right - I had this alias defined in .gitconfig:
```
  record = ! sh -c '(git add -p -- $@ && git commit) || git reset' --
```

It should be commit (in some form). If you are able to try it and test that it performs as required (reliable, no prompts) that would be very helpful.

**Comment by pfiaux** [comment]
I think calling `git add -p` isn't needed since the files are added in the for loop (L76). Also the patch option could result in prompts if some files were not added:
```
-p, --patch
   Interactively choose hunks of patch between the index and the work tree and add them to the
   index. This gives the user a chance to review the difference before adding modified contents to
   the index.

   This effectively runs add --interactive, but bypasses the initial command menu and directly jumps
   to the patch subcommand. See “Interactive mode” for details.
```

I think the following will do:
```
$GIT commit -m "$MSG" "$@" -- "$FILES" || $GIT reset
```

- Calls commit with a message and any optional arguments
- If commit fails unstage all the changes as your alias did (I don't think arguments are needed here)

I'll update my hledger-git and run some tests.

**Comment by simonmichael** [comment]
Fixed by #1949.
